providers/runtime.go

```diff
@@ -243,7 +229,7 @@ func (r *Runtime) Connect(req *plugin.ConnectReq) error {
 	}
 
 	r.Recording().EnsureAsset(r.Provider.Connection.Asset, r.Provider.Instance.ID, r.Provider.Connection.Id, asset.Connections[0])
-	r.schema.prioritizeIDs(BuiltinCoreID, r.Provider.Instance.ID)
+	// r.schema.prioritizeIDs(BuiltinCoreID, r.Provider.Instance.ID)
```

Comment:
we cannot really do this with a global schema. It seems like before we were prioritizing the IDs with the specific provider for the asset coming in first. I am not sure if that would actually break anything. @arlimus not sure if you recall why we needed that

Reply:
Yeah this one is going to break things if we don't keep it around...

The reason is that some keywords are equally implemented in two different providers (like `vulnmgmt` in os and vsphere) to provide the appropriate level of functionality in either, without having to learn new keywords. It is functionally correct, but it forces us to detect the right provider depending on the current prioritization...

(Long-term this is not a problem, because all keywords will be anchored to a parent, which entirely removes the guessword, but we can't force this into v10 right now...)

There are a few ways to handle this. First we need the list of resource definitions that have the same identifier. Once we have those, we can prioritize when we look up the resource/field based on the current connection. 

(a) We can introduce a new field into ResourceInfo that we can later easily remove (once everything is anchored to parents). We can't modify any part of ResourceInfo because it may be different between providers (which is ok). We could however add another field to it to chain things for now: `ResourceInfo.Other` which is either `nil` if no other provider offers it or is the next available info. 

(b) The alternative is to change the schema lookup from a simple `map[string]ResourceInfo` into a `map[string][]ResourceInfo`. It requires a bit more work and rewrite.

Reply:
I found 2 cases around this:
- 1 resource that can be defined in multiple providers (not extending, just having the same resource in multiple places)
- resource extensions where the root is in 1 place and the extensions are potentially in multiple providers

To cover both cases I picked approach (a) and added `Others` field that stores all other definitions of the `ResourceInfo`. However, I had to add the same `Others` field to the `Field` struct, because fields can also exist in multiple providers. Just like the `vulnmgmt` field that exists in the `vsphere` and `os` providers.

I think both cases work now. I still need to clean up the code a bit and will definitely add tests for both cases. Took me quite a bit of time to understand how this should work...

Reply:
Saw that you added others to cover fields as well, that's awesome!